src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/conditions/ExclusionZoneService.kt

```diff
@@ -2,91 +2,88 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.conditions
       .orElseThrow { EntityNotFoundException("$licenceId") }
-
-    val additionalCondition = additionalConditionRepository
-      .findById(conditionId)
+    val additionalCondition = additionalConditionRepository.findById(conditionId)
       .orElseThrow { EntityNotFoundException("$conditionId") }
 
-    // Remove any existing upload detail rows manually - intentionally not linked to the additionalCondition entity
-    // There can only be one uploaded exclusion map on this licence/condition
-    additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }.forEach {
-      additionalConditionUploadDetailRepository.findById(it).ifPresent { detail ->
-        additionalConditionUploadDetailRepository.delete(detail)
-      }
-    }
+    removeExistingDbRows(additionalCondition)
 
-    log.info("uploadExclusionZoneFile:  Name ${file.name} Type ${file.contentType} Original ${file.originalFilename}, Size ${file.size}")
+    log.info("uploadExclusionZoneFile: Name ${file.name} Type ${file.contentType} Original ${file.originalFilename}, Size ${file.size}")
 
-    if (file.isEmpty) {
-      log.error("uploadExclusion:  Empty file uploaded, Name ${file.name} Type ${file.contentType} Orig. Name ${file.originalFilename}, Size ${file.size}")
-      throw ValidationException("Exclusion zone - file was empty.")
+    val uploadFile = ExclusionZoneUploadFile(file)
+
+    if (uploadFile.isEmpty || null in listOf(uploadFile.fullSizeImage, uploadFile.thumbnailImage, uploadFile.description)) {
+      val reason = if (file.isEmpty) "Empty file" else "failed to extract the expected image map"
+      throw ValidationException("Exclusion zone upload document - $reason")
     }
 
-    // Process the MapMaker PDF file to get the fullSizeImage from page 1, descriptive text on page 2 and generate a thumbnail
-    val fullSizeImage = extractFullSizeImageJpeg(file.inputStream)
-    val description = extractDescription(file.inputStream)
-    val thumbnailImage = createThumbnailImageJpeg(fullSizeImage)
+    logAndUploadExclusionZoneFile(uploadFile = uploadFile, licence = licenceEntity, additionalCondition = additionalCondition)
+  }
 
-    // Validate that we were able to extract meaningful data from the uploaded file
-    if (fullSizeImage == null || thumbnailImage == null) {
-      log.error("uploadExclusion:  Could not extract images from file, Name ${file.name} Type ${file.contentType} Orig. Name ${file.originalFilename}, Size ${file.size}")
-      throw ValidationException("Exclusion zone - failed to extract the expected image map")
+  fun logAndUploadExclusionZoneFile(
+    uploadFile: ExclusionZoneUploadFile,
+    licence: Licence,
+    additionalCondition: AdditionalCondition,
+  ) {
+    val (originalDataDsUuid, fullSizeImageDsUuid, thumbnailImageDsUuid) = if (documentServiceEnabled) {
+      uploadDocuments(uploadFile = uploadFile, licence = licence, additionalCondition = additionalCondition)
+    } else {
+      Triple(null, null, null)
```

Comment:
could move this into the uploadDocuments so the toggle is encapsulated away, no body needs to know about it really, your going to remove it